.gitmodules

```diff
@@ -25,13 +25,14 @@
 [submodule "scipy/_lib/array_api_extra"]
 	path = scipy/_lib/array_api_extra
 	url = https://github.com/data-apis/array-api-extra.git
-[submodule "scipy/_lib/xsf"]
-	path = scipy/_lib/xsf
-	url = https://github.com/scipy/xsf.git
 # All submodules used as a Meson `subproject` are required to be under the
 # subprojects/ directory - see:
 # https://mesonbuild.com/Subprojects.html#why-must-all-subprojects-be-inside-a-single-directory
 [submodule "subprojects/highs"]
 	path = subprojects/highs
 	url = https://github.com/scipy/HiGHs
 	shallow = true
+[submodule "subprojects/xsf"]
+	path = subprojects/xsf
+	url = https://github.com/scipy/xsf.git
+	shallow = true
```

Comment:
can you avoid the `shallow` here? I think we only want that for repos with a very heavy commit history. There's not much benefit here I think, and it makes dev tasks more annoying sometimes (PRs won't check out because commit hash is unknown).